<launch>
	<!-- logger params-->
	<arg name="detection_rate"		default="50"		doc="Ros_rate in Hz"/>
	<arg name="object_name"			default="gapter1"	doc="Object name to capture in relevant topics for logging" />
	
	<arg name="do_debug"			default="false" 	doc="Print debug messages on output" />
	<arg name="do_preview"			default="false" 	doc="Show resulting image on screen" />
	<arg name="save_images"			default="true" 	doc="Save resulting image to folder" />
	<arg name="image_savepath"		default="$(env HOME)/projects/NE_UAScodes/logs/Leader_Follower/images_latest/"	doc="Path to save the images to"/>
	
	<arg name="num_rhombis"			default="3" 		doc="Maximum Number of Rhombis to be detected in the image" />  
	<arg name="self_rhombi"			default="0" 		doc="# of Rhombi fixed on the drone. Rhombi 0 has ids from 0 to 16. Rhombi 1 has ids from 16 to 33 etc. " />
	<arg name="min_detections"		default="2" 		doc="min # of marker detections per Rhombi to proceed with solvePnP" />
	
	<arg name="rhombi_gmtry_path"	default="$(find rhombi_markers)/cfg/Rhombicuboctahedron_V2" 	doc="Path for geometry to use" />>
	<arg name="use_prev_rvec_tvec"	default="false"		doc="Use previous rvec tvec as initial guess for solvePnP" />
	<arg name="do_fcu_refinement"	default="false" 	doc="Refine previous rvec tvec with new FCU pose" />
	<arg name="do_crop_image"		default="true" 		doc="Crop image for faster processing" />
	<arg name="crop_width"			default="1024" 		doc="Crop image for faster processing" />
	<arg name="crop_height"			default="768" 		doc="Crop image for faster processing" />

	<arg name="image_encoding"		default="mono8" 	doc="Image encoding" />
	<!-- ArUco parameters -->
	<arg name="DETECTION_MODE"		default="0" 		doc="ArUco detection mode: DM_NORMAL=0,DM_FAST=1,DM_VIDEO_FAST=2" />
	<arg name="REFINEMENT_MODE"		default="0" 		doc="ArUco Corner Refinement: CORNER_SUBPIX=0,CORNER_LINES=1,CORNER_NONE=2" />
	<arg name="DICTIONARY"			default="4" 		doc="ArUco dict to use: ARUCO_MIP_36h12=1, ARUCO=2, ARUCO_MIP_25h7=3, ARUCO_MIP_16h3=4 etc." />
	
	<arg name="contrast"			default="1.0" 		doc="Image contrast refinement" />
	<arg name="brightness"			default="0.0" 		doc="Image brightness refinement" />
	
	<arg name="cam_calib_file"		default="$(find spinnaker_sdk_camera_driver)/params/gapter1_matlab.yaml" doc="File specifying the parameters of the camera_array"/>
	
	<!-- load the logging node -->
	<node pkg="rhombi_markers" type="rhombi_detect" name="rhombi_detect_$(arg object_name)" output="screen" >
		<!-- Load parameters onto server using argument or default values above -->
		<rosparam command="load"        file="$(arg cam_calib_file)" />
		
		<param name="detection_rate"			value="$(arg detection_rate)" />
		<param name="object_name"			value="$(arg object_name)" />
		<param name="do_debug"				value="$(arg do_debug)" />
		<param name="do_preview"			value="$(arg do_preview)" />
		<param name="save_images"			value="$(arg save_images)" />
		<param name="num_rhombis"			value="$(arg num_rhombis)" />
		<param name="min_detections"		value="$(arg min_detections)" />
		<param name="image_savepath"		value="$(arg image_savepath)" />
		<param name="use_prev_rvec_tvec"	value="$(arg use_prev_rvec_tvec)" />
		<param name="do_fcu_refinement"		value="$(arg do_fcu_refinement)" />
		<param name="do_crop_image"			value="$(arg do_crop_image)" />
		<param name="crop_width"			value="$(arg crop_width)" />
		<param name="crop_height"			value="$(arg crop_height)" />
		<param name="image_encoding"		value="$(arg image_encoding)" />
		<param name="DETECTION_MODE"		value="$(arg DETECTION_MODE)" />
		<param name="REFINEMENT_MODE"		value="$(arg REFINEMENT_MODE)" />
		<param name="DICTIONARY"			value="$(arg DICTIONARY)" />
		<param name="self_rhombi"			value="$(arg self_rhombi)" />
		<param name="rhombi_gmtry_path"		value="$(arg rhombi_gmtry_path)" />
		<param name="contrast"				value="$(arg contrast)" />
		<param name="brightness"			value="$(arg brightness)" />
		<param name="cam_calib_file"		value="$(arg cam_calib_file)" />
  </node>
</launch>

